streaming/base/format/base/writer.py

```diff
@@ -44,8 +45,10 @@ class Writer(ABC):
             ``None``.
         hashes (List[str], optional): Optional list of hash algorithms to apply to shard files.
             Defaults to ``None``.
-        size_limit (int, optional): Optional shard size limit, after which point to start a new
-            shard. If ``None``, puts everything in one shard. Defaults to ``1 << 26``.
+        size_limit (Union[int, str], optional): Optional shard size limit, after which point to start a new
+            shard. If ``None``, puts everything in one shard. Can specify bytes
+            human-readable format as well, for example ``"100kb"`` for 100 kilobyte
+            (100*1024*1024) and so on. Defaults to ``1 << 26``.
```

Comment:
nit: `100*1024*1024` -> `100*1024` ? Also check all the occurrence.

Reply:
My bad. I'll be better about catching stuff like this, sorry!

Reply:
No problem. It happens!